src/core/algorithms/ind/spider/spider.cpp

```diff
@@ -0,0 +1,143 @@
+/** \file
+ * \brief Spider algorithm
+ *
+ * Spider algorithm class methods definition
+ */
+#include "spider.h"
+
+#include <functional>
+#include <memory>
+#include <queue>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#include "attribute.h"
+#include "config/equal_nulls/option.h"
+#include "config/error/option.h"
+#include "config/mem_limit/option.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/thread_number/option.h"
+#include "util/timed_invoke.h"
+
+namespace algos {
+
+using AttributeIndex = spider::AttributeIndex;
+
+Spider::Spider() : INDAlgorithm({}) {
+    DESBORDANTE_OPTION_USING;
+
+    RegisterOption(config::EqualNullsOpt(&is_null_equal_null_));
+    RegisterOption(config::ThreadNumberOpt(&threads_num_));
+    RegisterOption(config::MemLimitMBOpt(&mem_limit_mb_));
+    RegisterOption(config::ErrorOpt(&max_ind_error_));
+    MakeLoadOptsAvailable();
+}
+
+void Spider::MakeLoadOptsAvailable() {
+    MakeOptionsAvailable({config::EqualNullsOpt.GetName(), config::ThreadNumberOpt.GetName(),
+                          config::MemLimitMBOpt.GetName()});
+}
+
+void Spider::MakeExecuteOptsAvailable() {
+    MakeOptionsAvailable({config::ErrorOpt.GetName()});
+}
+
+void Spider::LoadDataInternal() {
+    auto create_domains = [&] {
+        domains_ = model::ColumnDomain::CreateFrom(input_tables_, mem_limit_mb_, threads_num_);
+    };
+    timings_.load = util::TimedInvoke(create_domains);
+}
+
+namespace {
+template <typename Attribute>
+std::vector<Attribute> InitAttributes(std::vector<model::ColumnDomain> const& domains) {
+    std::vector<Attribute> attrs;
+    AttributeIndex attr_count = domains.size();
+    attrs.reserve(attr_count);
+    for (AttributeIndex attr_id = 0; attr_id != attr_count; ++attr_id) {
+        attrs.emplace_back(attr_id, attr_count, domains[attr_id]);
+    }
+    return attrs;
+}
+
+template <typename Attribute>
+std::vector<Attribute> GetProcessedAttributes(std::vector<model::ColumnDomain> const& domains,
+                                              config::EqNullsType is_null_equal_null) {
+    using AttributeRW = std::reference_wrapper<Attribute>;
+    std::vector attrs = InitAttributes<Attribute>(domains);
+    AttributeIndex attr_count = attrs.size();
+    std::priority_queue<AttributeRW, std::vector<AttributeRW>, std::greater<Attribute>> attr_pq(
+            attrs.begin(), attrs.end());
+    while (!attr_pq.empty()) {
+        boost::dynamic_bitset<> ids_bitset(attr_count);
```

Comment:
How about move `ids_bitset` out of loop and use `reset` at the end of iteration?